main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -840,15 +840,19 @@ object BuildServerProtocol {
     }
   }
 
-  private def scalaTestClassesTask: Initialize[Task[ScalaTestClassesItem]] = Def.task {
-    val testClasses = Keys.definedTests.?.value
-      .getOrElse(Seq.empty)
-      .map(_.name)
-      .toVector
-    ScalaTestClassesItem(
-      bspTargetIdentifier.value,
-      testClasses
-    )
+  private def scalaTestClassesTask: Initialize[Task[Seq[ScalaTestClassesItem]]] = Def.task {
+    Keys.definedTests.?.value match {
+      case None => Vector.empty
+      case Some(defined) =>
+        defined.testPerFramework.map {
+          case (framework, tests) =>
+            ScalaTestClassesItem(
+              bspTargetIdentifier.value,
+              tests.map(_.name).toVector,
+              framework.name
```

Comment:
You cannot change the type of `Keys.definedTests`. Alternatively you can use `loadedTestFrameworks.value` to build a `Map[Fingerprint, Framework]` and then group each definedTests.value` by framework.

Reply:
I saw your comment too late :/ I've already tweaked my initial implementation to not break compatibility by just adding other keys but I find your proposition far better, I'll change that

Reply:
The only key which s changed now is `bspScalaTestClassesItem`. Is changing bsp keys acceptable or just to be fully compliant it's better to create a new one? 
cc: @eed3si9n

Reply:
Assuming no plugins are consuming `bsp*` keys, I am ok with making mima exceptions for those.

Reply:
`scala-debug-adapter` does not depend on `bspScalaTestClassesItem` and I found no consumer in https://github.com/search?p=1&q=bspScalaTestClassesItem&type=Code.

So I think we are good with this one.

Reply:
Sorry, I've never used MiMa before - if `mainProj/mimaReportBinaryIssues` passes should I change something with mima or is it ok?

Reply:
It is unclear to me why the `mimaReportBinaryIssues` passes. But since it does I think there is nothing else to do.

Reply:
Amina and I started on sbt 2.x on develop branch, so mima has been turned off. Maybe we should create 1.7.x branch off of 1.6.x?
https://github.com/sbt/sbt/blob/develop/.github/workflows/ci.yml#L102

Reply:
I just created [branch 1.7.x](https://github.com/sbt/sbt/tree/1.7.x). @kpodsiad you can repoen this PR to merge into 1.7.x.